pcdsdevices/delay_generator.py

```diff
@@ -0,0 +1,133 @@
+"""
+Delay generator class
+
+This module contains classes related to the SRS DG645 delay generator.
+"""
+
+from ophyd import Device
+from ophyd import Component as Cpt
+from ophyd import EpicsSignalRO
+from ophyd import EpicsSignal
+from ophyd.utils.epics_pvs import set_and_wait
+
+from .interface import BaseInterface
+
+
+CHANNELS = ['A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'T0']
+TRIGGER_SOURCES = {
+    '0': 'Internal',
+    '1': 'Ext ^edge',
+    '2': 'Ext ~edge',
+    '3': 'SS ext ^edge',
+    '4': 'SS ext ~edge',
+    '5': 'Single Shot',
+    '6': 'Line'
+}
+TRIGGER_INHIBITS = {
+    '0': 'Off',
+    '1': 'Triggers',
+    '2': 'AB',
+    '3': 'AB,CD',
+    '4': 'AB,CD,EF',
+    '5': 'AB,CD,EF,GH'
+}
+
+
+class DgChannel(BaseInterface, Device):
+    """
+    Delay generator single channel class
+
+    Parameters
+    ----------
+    prefix: str
+        Base PV for the delay generator channel
+
+    name: str
+        Alias of the channel
+    """
+    delay = Cpt(EpicsSignal, 'DelayAO', kind='hinted')
+    delay_str = Cpt(EpicsSignalRO, 'DelaySI', kind='normal')
+    reference = Cpt(EpicsSignal, 'ReferenceMO', kind='normal',
+                    doc='Reference channel from which the delay taken.')
+
+    tab_component_names = True
+    tab_whitelist = ['set', 'set_reference', 'get_str']
+
+    def get_str(self):
+        """ Returns the full <REF> + <DELAY> string. """
+        return self.delay_str.get()
+
+    def set(self, new_delay, **kwargs):
+        return self.delay.set(delay, **kwargs)
```

Comment:
This function has a typo, the arg is `new_delay` and the passed value is `delay`

Reply:
That'll teach me accepting random suggestions from strangers without double checking them :-p